py/torch_tensorrt/dynamo/conversion/impl/shuffle.py

```diff
@@ -101,41 +176,120 @@ def pixel_unshuffle(
     input: TRTTensor,
     downscale_factor: int,
 ) -> TRTTensor:
-    shape = input.shape
-    in_channels, in_height, in_width = shape[-3:]
-    out_channels = in_channels * (downscale_factor**2)
-    out_height = in_height // downscale_factor
-    out_width = in_width // downscale_factor
-    new_shape = shape[:-3] + (
-        in_channels,
-        out_height,
-        downscale_factor,
-        out_width,
-        downscale_factor,
-    )
-    reshaped_tensor = reshape(
-        ctx, target, source_ir, f"{name}_reshape1", input, new_shape
-    )
-    rank = len(new_shape)
-    permute_shape = tuple(range(rank - 5)) + (
+    # Get input shape tensor
+    input_shape_tensor = ctx.net.add_shape(input).get_output(0)
+
+    # Extract in_channels, in_height, and in_width from the input shape tensor
+    in_channels_tensor = ctx.net.add_slice(
+        input_shape_tensor, start=(len(input.shape) - 3,), shape=(1,), stride=(1,)
+    ).get_output(0)
+    in_height_tensor = ctx.net.add_slice(
+        input_shape_tensor, start=(len(input.shape) - 2,), shape=(1,), stride=(1,)
+    ).get_output(0)
+    in_width_tensor = ctx.net.add_slice(
+        input_shape_tensor, start=(len(input.shape) - 1,), shape=(1,), stride=(1,)
+    ).get_output(0)
+
+    # Calculate out_channels, out_height, and out_width as tensors
+    downscale_factor_sq = downscale_factor * downscale_factor
+    downscale_factor_tensor = get_trt_tensor(
+        ctx, downscale_factor, f"{name}_downscale_factor"
+    )
+    downscale_factor_sq_tensor = get_trt_tensor(
+        ctx, downscale_factor_sq, f"{name}_downscale_factor_sq"
+    )
+
+    out_channels_tensor = impl.elementwise.mul(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_channels_tensor",
+        in_channels_tensor,
+        downscale_factor_sq_tensor,
+    )
+    out_height_tensor = impl.elementwise.div(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_height_tensor",
+        in_height_tensor,
+        downscale_factor_tensor,
+    )
+    out_width_tensor = impl.elementwise.div(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_out_width_tensor",
+        in_width_tensor,
+        downscale_factor_tensor,
+    )
+
+    # Construct new shape tensor
+    new_shape_tensors = [
+        ctx.net.add_slice(
+            input_shape_tensor, start=(i,), shape=(1,), stride=(1,)
+        ).get_output(0)
+        for i in range(len(input.shape) - 3)
+    ]
+    new_shape_tensors += [
+        in_channels_tensor,
+        out_height_tensor,
+        get_trt_tensor(ctx, downscale_factor, f"{name}_downscale_factor_1"),
+        out_width_tensor,
+        get_trt_tensor(ctx, downscale_factor, f"{name}_downscale_factor_2"),
```

Comment:
Is there reason why get trt tensor again instead of reusing `downscale_factor_tensor` created by line 195? I think it would be faster if creates less ITensor in TRT

Reply:
Same as above.